go/adbc/driver/databricks/schema.go

```diff
@@ -0,0 +1,308 @@
+			arrowType = arrow.PrimitiveTypes.Float64
+		case colType == "boolean":
+			arrowType = arrow.FixedWidthTypes.Boolean
+		case colType == "timestamp":
+			arrowType = &arrow.TimestampType{Unit: arrow.Second}
+		case colType == "date":
+			arrowType = arrow.FixedWidthTypes.Date32
+		case strings.HasPrefix(colType, "decimal"):
+			// Parse decimal precision and scale from format "decimal(precision,scale)"
+			if matches := regexp.MustCompile(`decimal\((\d+),(\d+)\)`).FindStringSubmatch(colType); matches != nil {
+				precision, _ := strconv.ParseInt(matches[1], 10, 32)
+				scale, _ := strconv.ParseInt(matches[2], 10, 32)
+				arrowType = &arrow.Decimal256Type{Precision: int32(precision), Scale: int32(scale)}
+			} else {
+				arrowType = &arrow.Decimal256Type{}
+			}
+		default:
+			err := fmt.Errorf("unsupported type: %v", colType)
+			return nil, err
+		}
+		fields[i] = arrow.Field{
+			Name:     col["name"].(string),
+			Type:     arrowType,
+			Nullable: true,
+			Metadata: arrow.MetadataFrom(map[string]string{
+				MetadataKeyDatabricksType: colType,
+			}),
+		}
+	}
+	// TODO: include relevant metadata from dbrx into the Arrow schema
+	return arrow.NewSchema(fields, nil), nil
+}
+
+func ResultSchemaToArrowSchema(dbx_schema *sql.ResultSchema) (*arrow.Schema, error) {
+	fields := make([]arrow.Field, dbx_schema.ColumnCount)
+	for i, col := range dbx_schema.Columns {
+		arrowType, err := getArrowTypeFromColumnInfo(col)
+		if err != nil {
+			return nil, err
+		}
+		fields[i] = arrow.Field{
+			Name:     col.Name,
+			Type:     arrowType,
+			Nullable: true,
+			Metadata: arrow.MetadataFrom(map[string]string{
+				MetadataKeyDatabricksType: string(col.TypeName),
+				"position":                col.Name,
+				"type_text":               col.TypeText,
+			}),
+		}
+	}
+	metadata := arrow.MetadataFrom(map[string]string{
+		"column_count": strconv.Itoa(dbx_schema.ColumnCount),
+	})
+	return arrow.NewSchema(fields, &metadata), nil
+}
+
+func getArrowTypeFromColumnInfo(col sql.ColumnInfo) (arrow.DataType, error) {
+	if arrowType, ok := basicTypeToArrowTypeMap[col.TypeName]; ok {
+		return arrowType, nil
+	}
+	switch col.TypeName {
+	case sql.ColumnInfoTypeNameDecimal:
+		precision, scale := col.TypePrecision, col.TypeScale
+		if precision == 0 {
+			// Mostly used by the recursive case (array, map, etc)
+			// "DECIMAL(p,s) and DECIMAL(p) with scale default at 0"
+			if matches := regexp.MustCompile(`DECIMAL\((\d+)(?:,?(\d+))?\)`).FindStringSubmatch(col.TypeText); matches != nil {
+				var err error
+				precision, err = strconv.Atoi(matches[1])
+				if err != nil {
+					return nil, fmt.Errorf("invalid decimal precision: %v", err)
+				}
+				if len(matches) == 2 {
+					scale, err = strconv.Atoi(matches[2])
+					if err != nil {
+						return nil, fmt.Errorf("invalid decimal scale: %v", err)
+					}
+				}
+			}
+		}
+		if precision <= 38 {
+			return &arrow.Decimal128Type{Precision: int32(precision), Scale: int32(scale)}, nil
+		}
```

Comment:
We need to ensure the `cmd_record_reader.go` can process all these types correctly. The data resulting from the processing of the result set should perfectly match the types that were chosen here in `schema.go`.